tests/mock_network/node_ageing.rs

```diff
@@ -0,0 +1,228 @@
+// Copyright 2019 MaidSafe.net limited.
+//
+// This SAFE Network Software is licensed to you under The General Public License (GPL), version 3.
+// Unless required by applicable law or agreed to in writing, the SAFE Network Software distributed
+// under the GPL Licence is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied. Please review the Licences for the specific language governing
+// permissions and limitations relating to use of the SAFE Network Software.
+
+use super::{
+    add_connected_nodes_until_one_away_from_split, create_connected_nodes_until_split,
+    current_sections, poll_and_resend_with_options, PollOptions, TestNode, MIN_SECTION_SIZE,
+};
+use rand::Rng;
+use routing::{mock::Network, FullId, NetworkConfig, Prefix, XorName};
+use std::{iter, slice};
+
+#[test]
+fn relocate_without_split() {
+    // Create a network of two sections, then trigger relocation of a random node from one section
+    // into the other section.
+    let network = Network::new(MIN_SECTION_SIZE, None);
+    let mut rng = network.new_rng();
+    let mut nodes = create_connected_nodes_until_split(&network, vec![1, 1]);
+
+    let prefixes: Vec<_> = current_sections(&nodes).collect();
+    let source_prefix = *unwrap!(rng.choose(&prefixes));
+    let target_prefix = *choose_other_prefix(&mut rng, &prefixes, &source_prefix);
+
+    let relocate_index = choose_node_from_section(&mut rng, &nodes, &source_prefix);
+    let relocate_id = nodes[relocate_index].id();
+
+    let destination = gen_name_in_prefix(&mut rng, &target_prefix);
+
+    // Trigger relocation.
+    for node in nodes
+        .iter_mut()
+        .filter(|node| source_prefix.matches(&node.name()))
+    {
+        // TODO: When relocation trigger is implemented, change this test to use it instead of this
+        // explicit method.
```

Comment:
The relocation trigger could be implemented already. `Chain::increase_members_age` could check if some members' age counters are exact powers of 2, and return their IDs, which would then be returned from `Chain::add_member` and `Chain::remove_member` - these IDs would be the ones to be relocated.

Reply:
That's exactly what the current plan is, except that it's going to be implemented in a separate PR.